src/zinolib/config/zino1.py

```diff
@@ -0,0 +1,57 @@
+from pydantic import BaseModel
+
+from . import tcl
+from .models import UserConfig, ServerV1Config, Options
+
+
+def _parse_tcl(config_dict, section):
+    fixed_dict = tcl.normalize(config_dict)
+    connection = fixed_dict["connections"][section]
+    options = fixed_dict["globals"]
+    connection['password'] = connection.pop("secret")
+    return connection, options
+
+
+class ZinoV1Config(UserConfig, ServerV1Config, Options):
+    """
+    How to use::
+
+    Make a config-class from the tcl-config stored on disk::
+
+        > config = ZinoV1Config.from_tcl()
+
+    Get the actual user and Zino1 secret and update the config-object::
+
+        > config.set_userauth(actual_username, secret)
+
+    Read some command-line arguments via argparse.ArgumentParser and update the
+    config::
+
+        > config.update_from_args(args)
+    """
+
+    @classmethod
+    def from_tcl(cls, filename=None, section="default"):
+        config_dict = tcl.parse_tcl_config(filename)
+        connection, options = _parse_tcl(config_dict, section)
+        return cls(**connection, **options)
+
+    @classmethod
+    def from_toml(cls, filename=None, section="default"):
+        config_dict = toml.parse_toml_config(filename)
+        return cls(**config_dict["connections"], **config_dict["options"])
+
+    def set_userauth(self, username, password):
+        self.username = username
+        self.password = password
+
+    def update_from_args(self, args):
+        """
+        Assumes argparse-style args namespace object
+
+        arg-names not found in the config-object are ignored.
+        """
```

Comment:
Why use docstrings when there are type annotations available?
```suggestion
    def update_from_args(self, args: argparse.Namespace):
        """Updates the configuration from command line arguments.
        
        arg-names not found in the config-object are ignored.
        """
```

Reply:
Because I consider annotations out-of-scope before the "API" is locked in. This is also why I've marked your suggested annotations as resolved. Other code is waiting for this code, and I want some feedback on use before I start polishing and locking things down. This is not the final form of these files: lots more planned, but it needs merging so there is less branch-juggling.

Reply:
If docstrings can change, so can type annotations.